posts/2021-10-05-configurable-response-headers-21.0.0.11-beta.adoc

```diff
@@ -0,0 +1,182 @@
+* <<jakarta, Jakarta EE 9.1 Beta Features>>: a lightweight package that contains only the Jakarta EE 9.1 features.
+
+This means that you can now try out our in-development Open Liberty features by just adding the relevant coordinates to your build tools.
+
+If you try either package, <<feedback, let us know what you think>>.
+[#allbeta]
+== All Beta Features package
+
+The All Beta Features package includes the following beta features:
+
+* <<response_headers, Configurable Response Headers>>
+
+[#response_headers]
+=== Configurable Response Headers
+
+You can now configure Open Liberty (21.0.0.11-beta) to modify response headers. 
+The configuration options enable you to:
+
+* append headers;
+* overwrite existing headers;
+* add missing headers; and
+* remove undesired headers from all responses that are serviced by an HTTP endpoint.
+
+This granular control over response headers means you can modify headers without having to modify existing applications or filters.
+
+==== Try it out! 
+
+To use configurable response headers, begin by defining a new element called `headers` in the `server.xml`.  You can configure this for individual HTTP endpoints or for all endpoints at once. 
+
+Configuring for individual HTTP endpoints:
+
+[source, xml]
+----
+<httpEndpoint id="defaultHttpEndpoint"
+                  httpPort="9080"
+                  httpsPort="9443" >
+	 
+       <headers>
+            <add>foo:bar</add>
+            <add>foo:bar2</add>
+            <set>customHeader:customValue</set>
+            <setIfMissing>X-Forwarded-Proto:https</setIfMissing>
+            <remove>Via</remove>
+       </headers>
+   		
+</httpEndpoint>
+----
+
+Configuring for all HTTP endpoints:
+
+[source, xml]
+----
+ <httpEndpoint id="defaultHttpEndpoint"
+                         httpPort="9080"
+                         httpsPort="9443"
+                         headersRef="myHeadersID">
+     </httpEndpoint>
+ 
+     <httpEndpoint id="otherHttpEndpoint"
+                         httpPort="9081"
+                         httpsPort="9444"
+                         headersRef="myHeadersID">
+    </httpEndpoint> 
+     
+     <headers id="myHeadersID">
+            <add>foo:bar</add>
+            <add>foo:bar2</add>
+            <set>customHeader:customValue</set>
+            <setIfMissing>X-Forwarded-Proto:https</setIfMissing>
+            <remove>Via</remove>
+     </headers>  
+----
+
+The `add` attribute allows multiple headers with the same name to be added to a response, similar to the HttpServletResponse's `addHeader` API. Similarly, the `set` attribute is analogous to the `setHeader` API, which sets a response header to the given name and value. This overwrites existing headers that share the same name. The setIfMissing attribute sets the configured headers only if they are not already present on the response. Lastly, the remove attribute removes any response headers with a name that matches the value that is specified in the attribute.
```

Comment:
the `setIfMissing` and `remove` words should also be highlighted like `add` and `set` are highlighted.  It is the two that are before the word `attribute`.

Reply:
```suggestion
The `add` attribute allows multiple headers with the same name to be added to a response, similar to the HttpServletResponse's `addHeader` API. Similarly, the `set` attribute is analogous to the `setHeader` API, which sets a response header to the given name and value. This overwrites existing headers that share the same name. The `setIfMissing` attribute sets the configured headers only if they are not already present on the response. Lastly, the `remove` attribute removes any response headers with a name that matches the value that is specified in the attribute.
```